phoenix-core/src/it/java/org/apache/phoenix/end2end/salted/SaltedTableMergeBucketsIT.java

```diff
@@ -0,0 +1,103 @@
+        statement.execute("CREATE TABLE " + tableName
+                + " (c1 VARCHAR NOT NULL, c2 VARCHAR NOT NULL, c3 VARCHAR NOT NULL,"
+                + " CONSTRAINT pk PRIMARY KEY(c1,c2,c3)) SALT_BUCKETS=11");
+        statement.execute(
+            " upsert into " + tableName + " values('20191211','HORTONWORKS_WEEKLY_TEST','v3')");
+        statement.execute(
+            " upsert into " + tableName + " values('20191212','HORTONWORKS_WEEKLY_TEST','v3')");
+        statement.execute(
+            " upsert into " + tableName + " values('20191213','HORTONWORKS_WEEKLY_TEST','v3')");
+        statement.execute(
+            " upsert into " + tableName + " values('20191214','HORTONWORKS_WEEKLY_TEST','v3')");
+        statement.execute(
+            " upsert into " + tableName + " values('20191215','HORTONWORKS_WEEKLY_TEST','v3')");
+        statement.execute(
+            " upsert into " + tableName + " values('20191216','HORTONWORKS_WEEKLY_TEST','v3')");
+        statement.execute(
+            " upsert into " + tableName + " values('20191217','HORTONWORKS_WEEKLY_TEST','v3')");
+        statement.execute(
+            " upsert into " + tableName + " values('20191218','HORTONWORKS_WEEKLY_TEST','v3')");
+        statement.execute(
+            " upsert into " + tableName + " values('20191219','HORTONWORKS_WEEKLY_TEST','v3')");
+        statement.execute(
+            " upsert into " + tableName + " values('20191220','HORTONWORKS_WEEKLY_TEST','v3')");
+        statement.execute(
+            " upsert into " + tableName + " values('20191221','HORTONWORKS_WEEKLY_TEST','v3')");
+        statement.execute(
+            " upsert into " + tableName + " values('20191222','HORTONWORKS_WEEKLY_TEST','v3')");
+        statement.execute(
+            " upsert into " + tableName + " values('20191223','HORTONWORKS_WEEKLY_TEST','v3')");
+        statement.execute(
+            " upsert into " + tableName + " values('20191224','HORTONWORKS_WEEKLY_TEST','v3')");
+        statement.execute(
+            " upsert into " + tableName + " values('20191225','HORTONWORKS_WEEKLY_TEST','v3')");
+        connection.commit();
+
+        Admin admin =
+                driver.getConnectionQueryServices(getUrl(), TestUtil.TEST_PROPERTIES).getAdmin();
+        List<RegionInfo> regions = admin.getRegions(TableName.valueOf(tableName));
+        for (int i = 0; i < regions.size() - 1; i+=2) {
+            admin.mergeRegionsAsync(regions.get(i).getEncodedNameAsBytes(),
+                regions.get(i + 1).getEncodedNameAsBytes(), false).get();
+        }
+        ResultSet rs =
+                statement.executeQuery("select c1,c2,c3 from " + tableName
+                        + " where c1='20191211' and c2 like '%HORTONWORKS_WEEKLY_TEST%'");
+        assertTrue(rs.next());
+        rs =
+                statement.executeQuery("select c1,c2,c3 from " + tableName
+                        + " where c1='20191217' and c2 like '%HORTONWORKS_WEEKLY_TEST%'");
```

Comment:
shall we pick random c1 values from a list of values instead of only with `20191211` and `20191217`?

we can create a list with multiple values and try using UPSERT in a loop against the list, that will reduce num of unique UPSERT statements we need to run in the test and also facilitate querying against any random c1 values from the given list.

Reply:
Random values in tests are fishy.
We could increase the number of rows, and check that we get back each of them.
It's a bit tricky, as the nature of hashing makes it hard to make sure that we cover all buckets.

Reply:
The now generates and checks 100 rows, that should be enough to hit all buckets (statistically).